frame/tips/src/tests.rs

```diff
@@ -286,6 +286,30 @@ fn close_tip_works() {
 	});
 }
 
+#[test]
+fn slash_tip_works() {
+	new_test_ext().execute_with(|| {
+		System::set_block_number(1);
+		Balances::make_free_balance_be(&Treasury::account_id(), 101);
+		assert_eq!(Treasury::pot(), 100);
+		assert_eq!(Balances::reserved_balance(0), 0);
+		assert_eq!(Balances::free_balance(0), 100);
+		assert_ok!(TipsModTestInst::report_awesome(Origin::signed(0), b"awesome.dot".to_vec(), 3));
+		assert_eq!(Balances::reserved_balance(0), 12);
+		assert_eq!(Balances::free_balance(0), 88);
+		let h = tip_hash();
+		assert_eq!(last_event(), RawEvent::NewTip(h));
+		assert_noop!(
+			TipsModTestInst::slash_tip(Origin::signed(0), h.clone()),
+			BadOrigin,
+		);
+		assert_ok!(TipsModTestInst::slash_tip(Origin::root(), h.clone()));
+		assert_eq!(last_event(), RawEvent::TipSlashed(h, 0, 12));
+		assert_eq!(Balances::reserved_balance(0), 0);
+		assert_eq!(Balances::free_balance(0), 88);
```

Comment:
```suggestion
		System::set_block_number(1);
		Balances::make_free_balance_be(&Treasury::account_id(), 101);
		assert_eq!(Treasury::pot(), 100);
		
		assert_eq!(Balances::reserved_balance(0), 0);
		assert_eq!(Balances::free_balance(0), 100);
		
		assert_ok!(TipsModTestInst::report_awesome(Origin::signed(0), b"awesome.dot".to_vec(), 3));
		
		assert_eq!(Balances::reserved_balance(0), 12);
		assert_eq!(Balances::free_balance(0), 88);
		
		let h = tip_hash();
		assert_eq!(last_event(), RawEvent::NewTip(h));
		
		// can't remove from any origin
		assert_noop!(
			TipsModTestInst::slash_tip(Origin::signed(0), h.clone()),
			BadOrigin,
		);
		
		// can remove from root.
		assert_ok!(TipsModTestInst::slash_tip(Origin::root(), h.clone()));
		assert_eq!(last_event(), RawEvent::TipSlashed(h, 0, 12));
		
		// tipper slashed
		assert_eq!(Balances::reserved_balance(0), 0);
		assert_eq!(Balances::free_balance(0), 88);
```

proper use of spaces can make your code much more _readable_,  without the need for much comments.